pageserver/src/page_service.rs

```diff
@@ -273,10 +273,28 @@ async fn page_service_conn_main(
                 info!("Postgres client disconnected ({io_error})");
                 Ok(())
             } else {
-                Err(io_error).context("Postgres connection error")
+                let tenant_id = conn_handler
+                    .timeline_handles
+                    .tenant_id()
+                    .map(|tid| format!("{tid}"))
+                    .unwrap_or("???".to_string());
+                Err(io_error).context(format!(
+                    "Postgres connection error for tenant {} client at {}",
+                    tenant_id, peer_addr
+                ))
             }
         }
-        other => other.context("Postgres query error"),
+        other => {
+            let tenant_id = conn_handler
+                .timeline_handles
+                .tenant_id()
+                .map(|tid| format!("{tid}"))
+                .unwrap_or("???".to_string());
+            other.context(format!(
+                "Postgres query error for tenant {} client at {}",
+                tenant_id, peer_addr
```

Comment:
could we instead simulate the normal tracing logging (which we don't have here for the unfortunate retuning of the error):

```suggestion
                "Postgres query error for tenant_id={} client at peer_addr={}",
                tenant_id, peer_addr
```

This suggestion for the both places of this top level context.

Reply:
Sure: https://github.com/neondatabase/neon/pull/9130/commits/a1c568098c3b2d8157fc75cce3aaff65a7657730